astrophot/param/param_context.py

```diff
@@ -0,0 +1,67 @@
+from .base import Node
+
+__all__ = ("Param_Unlock", "Param_SoftLimits", "Param_Mask")
+
+class Param_Unlock:
+    """Temporarily unlock a parameter.
+
+    Context manager to unlock a parameter temporarily. Inside the
+    context, the parameter will behave as unlocked regardless of its
+    initial condition. Upon exiting the context, the parameter will
+    return to it's previous locked state regardless of any changes
+    made by the user to the lock state.
+
+    """
+
+    def __init__(self, param = None):
+        self.param = param
+
+    def __enter__(self):
+        if self.param is None:
+            Node.global_unlock = True
+        else:
+            self.original_locked = self.param.locked
+            self.param.locked = False
+            
+    def __exit__(self, *args, **kwargs):
+        if self.param is None:
+            Node.global_unlock = False
+        else:
+            self.param.locked = self.original_locked
+
+class Param_SoftLimits:
+    """
+    Temporarily allow writing parameter values outside limits.
+
+    Values outside the limits will be quietly shift until they are within the boundaries of the parameter limits.
```

Comment:
"quietly shifted"?
Which boundary will they be shifted to in the case of cyclical constraints, such as PA?

Reply:
"quietly shifted" just means no errors or warnings are thrown. For cyclical constraints there is no need to shift, I just use modulo `%` to wrap it back into the range. I'll add to the docs